cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/TransformDetail.java

```diff
@@ -21,30 +21,45 @@
 
 /**
  * Class that encapsulates {@link co.cask.cdap.etl.api.Transform} and transformId
- * and {@link DefaultStageMetrics}
+ * and {@link DefaultStageMetrics} and boolean to indicate if the transform is sink.
  */
 public class TransformDetail {
 
   private final String transformId;
   private final Transformation transformation;
   private final StageMetrics metrics;
+  private final boolean isSink;
 
-  public TransformDetail(String transformId, Transformation transform, StageMetrics metrics) {
+  public TransformDetail(String transformId, Transformation transform, StageMetrics metrics, boolean isSink) {
     this.transformation = transform;
     this.transformId = transformId;
     this.metrics = metrics;
+    this.isSink = isSink;
+  }
+
+  public TransformDetail(String transformId, StageMetrics metrics, boolean isSink) {
+    this(transformId, null, metrics, isSink);
+  }
+
+  public TransformDetail(String transformId, Transformation transform, StageMetrics metrics) {
+    this(transformId, transform, metrics, false);
   }
 
   public TransformDetail(TransformDetail transformDetail, Transformation transformation) {
     this.transformation = transformation;
     this.transformId = transformDetail.getTransformId();
     this.metrics = transformDetail.getMetrics();
+    this.isSink = transformDetail.isSink();
   }
 
   public Transformation getTransformation() {
     return transformation;
   }
 
+  public boolean isSink() {
```

Comment:
Why need this? It's unnatural to have transformation detail need to know whether it is a sink or not as it seems totally unrelated. Is there anywhere documenting the principle of the implementation?